site.py

```diff
@@ -0,0 +1,86 @@
+import os
+import subprocess
+import urllib
+import json
+import time
+import random
+
+from flask import Flask, render_template, url_for, request, redirect, send_from_directory
+
+
+ALLOWED_EXTENSIONS = set(['csv'])
+app = Flask(__name__)
+app.config['UPLOAD_FOLDER'] = 'tmp/'+ str(random.randint(10000, 99999))+'/'
+
+# Confirm that a provided file has an acceptable extension.
+# In this case, '.csv' is the only allowed extension.
+def allowed_file(filename):
+    return ('.' in filename) and (filename.rsplit('.', 1)[1] in ALLOWED_EXTENSIONS)
+
+# Retrieve the files uploaded by the user, and if they are valid,
+#  store them in a new generated subfolder of tmp/.
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        metadatafile = request.files['metadatafile']
+        datafile = request.files['datafile']
+        
+        if metadatafile and allowed_file(metadatafile.filename) \
```

Comment:
Break up this condition to check for the erroneous cases. Render the template with a parameter that can be checked to toggle error messages.

```python
if not metadatafile or not datafile:
    return render_template('index.html', missing_fields=True)

if not allowed_file(metadatafile.filename) or not allowed_file(datafile.filename):
    return render_template('index.html', wrong_filetype=True)

# Normal processing...
```